bfs.go

```diff
@@ -56,3 +59,36 @@ type Iterator interface {
 type supportsCopying interface {
 	Copy(context.Context, string, string) error
 }
+
+// --------------------------------------------------------------------
+
+// Resolve opens a bucket from a URL string.
```

Comment:
Minor: doc contradicts with impl - it's not a "url string", but already parsed URL.

Reply:
Also, the goal is to save some repetitive code, so accepting a string would be more convenient.

Reply:
@mxmCherry I thought about passing a string first, but that's not really helpful if you want to use the URL. `Resolve` would have to parse the string and return the `URL` which is an inconvenience and not exactly what I was after with "minimal"